### DISCUSSION TITLE
Include an `.a` file from a cpp library

### DISCUSSION THREAD

**Discussion by KevinBLT** [vlang/v#19548] [Questions and Answers] [answered]
Hello everyone,

I am new to vlang and interested in using it - looks very good so far!

I found a project: https://github.com/jinja2cpp/Jinja2Cpp and built it,
but is created `.a` file instead of `.o`, would I still be able to use/import that into
v? Or how would I build it correctly for that? I have no idea about cmake and all that stuff that actually happens in there.

Generally, it's about reusing jinja2 templates with v http server. We need this generally for "every language", so html templates and such are not depending on the language (you can use those templates, with php, js, go, rust, ruby, python, ... you see the pattern).

Would anybody be able to help or would it generally be possible to do it like this (instead of `.o`/ `.c`):

```v
#flag -I @VMODROOT/Jinja2Cpp
#flag @VMODROOT/Jinja2Cpp/build/libjinja2cpp.a
#include "build/dist/include/jinja2cpp/template.h"

fn main() {
	// ... stuff going on
}
```

**Comment by JalonSolov** [comment, marked as answer]
See https://github.com/vlang/v/blob/master/doc/docs.md#passing-c-compilation-flags
